Feature/callbacks

RomanBachaloSigmaSoftware opened the pull request:
- Added functions with callbacks
- Added routes for geocode and forecast

RomanBachaloSigmaSoftware committed:
Initialization folder structure
- Distributed files in the appropriate folders
- Added routes.js for all routings
- Added fs.js for filesystem exercises endpoints
- Added fs-app.js with yargs commands, npm's request and chalk packages

RomanBachaloSigmaSoftware committed:
Added functions with callbacks

RomanBachaloSigmaSoftware committed:
Merge branch 'feature/folder-structure' into feature/callbacks

RomanBachaloSigmaSoftware committed:
Added geocode and forecast to routes

RomanBachaloSigmaSoftware committed:
Added forecast by place

Vadym803 commented on src/callbacks/app.js:
```diff
@@ -0,0 +1,39 @@
+]
+
+// gets the coordinates by given address and calls a callback function
+const weatherForecast = (address, callback) => {
```
1. Make this method pure (for ex., pass mapboxUrl as an argument)
2. Add validation for params you used. Is mapboxUrl[0] or mapboxUrl[1] really exists? Y
You can use lodash library for js and use methods like this
`_.has({ features: [{ center: [4,5,6] }, { center: [7,8,9] }] }, 'features[0].center[0]')`

Vadym803 commented on src/callbacks/app.js:
```diff
@@ -0,0 +1,39 @@
+
+// gets the weather for given coordinates
+const getWeather = (coordinates) => {
+    request({ url: weatherUrl[0] + coordinates.latitude + weatherUrl[1] + coordinates.longitude + weatherUrl[2], json: true }, (err, res) => {
```
Don't use string concatenation with "+".
Use `${someVar}${someAnotherVar` instead

Vadym803 commented on src/fs-app/index.js:
```diff
@@ -0,0 +1,111 @@
+        let url = baseUrl + argv.name
+        request({ url: url }, (err, res) => {
+            if (err) {
+                console.log(chalk.red(err))
```
Create a logger service and use methods from them.

For this application you can wrap log method with code

```
log(logData) {
  console.log(logData);
}
```

We need this because if we plan to change log destination (for example, from console log to to database) we don't need to change all places in the code anymore

Vadym803 commented on src/routes/forecast.js:
```diff
@@ -0,0 +1,38 @@
+            request({ url: `http://localhost:${port}/forecast/${geoResponse.body.latitude}/${geoResponse.body.longitude}`, json: true }, (error, response) => {
+                if (error) {
+                    console.log(error)
+                } else {
```
This functions is so easy and you can define it in another place and reuse in some place.

Vadym803 commented on src/routes/fs.js:
```diff
@@ -22,16 +16,18 @@ app.delete('/:fileName', (req, res) => {
     let file = `${documentValues.fileName}.${documentValues.fileExtension}`
     fs.exists(file, exists => {
         if (exists) {
+            res.statusCode = 400
```
Use constant as here
https://github.com/RomanBachaloSigmaSoftware/nodejs-practise/pull/1/commits/1ba428bf23d30bd8d572f9dbd50f7713316772f1#diff-83e5c273ee2c267ac0248fcbe54965012c98a0105c210136b7d902e9ca4ff594R19

Vadym803 commented on src/routes/routes.js:
```diff
@@ -0,0 +1,13 @@
+const app = express()
+const port = 3000
+
+app.use('/fs', express.raw({ type: "application/json" }), fsRoute)
```
You don't need to copy paste which parser to use.

use one line instead
`app.use(bodyParser.json({ limit: '64kb' })); // for parsing application/json`

RomanBachaloSigmaSoftware committed:
Fix after review
- Added logService
- Added constants
- Added parameters validation
- Changed + string concatenation with ${} constructions

RomanBachaloSigmaSoftware committed:
Added forecastCallback

RomanBachaloSigmaSoftware committed:
Added semicolons where needed
